### ISSUE TITLE
[BUG] Possible endlessly hanging threads in Async clients

### ISSUE THREAD

**Issue by TheDevOps** [Azure/azure-sdk-for-java#13903]
**Describe the bug**
"com.microsoft.azure.servicebus.management.ManagementClientAsync.getSubscriptionRuntimeInfoAsync(String, String)" can end up with an endlessly stuck thread in case the parsing of the loaded xml content causes a runtime exception. From a quick code check this is the case for pretty much all async methods right now, this is just the one where we encountered the issue.

***Exception or Stack Trace***
There is no exception or stacktrace logged right now the only thing is that the CompletableFuture returned by the method never returns, a stack dump shows for example

```
"http-nio-8453-exec-7" #119 daemon prio=5 os_prio=0 tid=0x0000558c0ebde000 nid=0xbe waiting on condition [0x00007fb43028b000]
   java.lang.Thread.State: WAITING (parking)
	at sun.misc.Unsafe.park(Native Method)
	- parking to wait for  <0x00000000db56c0c8> (a java.util.concurrent.CompletableFuture$Signaller)
	at java.util.concurrent.locks.LockSupport.park(LockSupport.java:175)
	at java.util.concurrent.CompletableFuture$Signaller.block(CompletableFuture.java:1693)
	at java.util.concurrent.ForkJoinPool.managedBlock(ForkJoinPool.java:3323)
	at java.util.concurrent.CompletableFuture.waitingGet(CompletableFuture.java:1729)
	at java.util.concurrent.CompletableFuture.get(CompletableFuture.java:1895)
	at com.microsoft.azure.servicebus.Utils.completeFuture(Utils.java:17)
	at com.microsoft.azure.servicebus.management.ManagementClient.getSubscriptionRuntimeInfo(ManagementClient.java:144)
```

and remains in this state until the application is terminated.

**To Reproduce**
Reproductions with real code is a bit tricky since right now it depends on the XML returned by the server being invalid, for this a separate issue with azure will be filed however see the code snipped for a quick test code that explains the problem

***Code Snippet***
The actual problems happens with

```
    public CompletableFuture<SubscriptionDescription> getSubscriptionAsync(String topicPath, String subscriptionName) {
        EntityNameHelper.checkValidTopicName(topicPath);
        EntityNameHelper.checkValidSubscriptionName(subscriptionName);

        String path = EntityNameHelper.formatSubscriptionPath(topicPath, subscriptionName);
        CompletableFuture<String> contentFuture = getEntityAsync(path, null, false);
        CompletableFuture<SubscriptionDescription> sdFuture = new CompletableFuture<>();
        contentFuture.handleAsync((content, ex) -> {
            if (ex != null) {
                sdFuture.completeExceptionally(ex);
            } else {
                try {
                    sdFuture.complete(SubscriptionDescriptionSerializer.parseFromContent(topicPath, content));
                } catch (MessagingEntityNotFoundException e) {
                    sdFuture.completeExceptionally(e);
                }
            }
            return null;
        }, MessagingFactory.INTERNAL_THREAD_POOL);

        return sdFuture;
    }
```

where due to an invalid xml response from the server "SubscriptionDescriptionSerializer.parseFromContent(topicPath, content)" throws a

```
java.time.format.DateTimeParseException: Text '2020-07-29T08:24:51.229' could not be parsed at index 23
    at java.time.format.DateTimeFormatter.parseResolved0(DateTimeFormatter.java:1949)
    at java.time.format.DateTimeFormatter.parse(DateTimeFormatter.java:1851)
    at java.time.Instant.parse(Instant.java:395)
    at com.microsoft.azure.servicebus.management.SubscriptionRuntimeInfoSerializer.parseFromEntry(SubscriptionRuntimeInfoSerializer.java:67)
    at com.microsoft.azure.servicebus.management.SubscriptionRuntimeInfoSerializer.parseFromContent(SubscriptionRuntimeInfoSerializer.java:33)
    at com.microsoft.azure.servicebus.management.ManagementClientAsync.lambda$getSubscriptionRuntimeInfoAsync$6(ManagementClientAsync.java:279)
    at java.util.concurrent.CompletableFuture.uniHandle(CompletableFuture.java:836)
    at java.util.concurrent.CompletableFuture$UniHandle.tryFire(CompletableFuture.java:811)
    at java.util.concurrent.CompletableFuture$Completion.run(CompletableFuture.java:456)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
    at java.lang.Thread.run(Thread.java:748)
```

Since this is never caught there is never any .completeXXX called on the returned CompleteableFuture and so it hangs endlessly. A quick example to easier reproduce this is

```
    public static void main(String[] args) throws ServiceBusException, InterruptedException, ExecutionException
    {
        CompletableFuture<String> contentFuture = CompletableFuture.completedFuture("TEST");
        CompletableFuture<String> sdFuture = new CompletableFuture<>();
        contentFuture.handleAsync((content, ex) -> {
            if (ex != null) {
                sdFuture.completeExceptionally(ex);
            } else {
                try {
                    sdFuture.complete(getTestString());
                } catch (MessagingEntityNotFoundException e) {
                    sdFuture.completeExceptionally(e);
                }
            }
            return null;
        }, MessagingFactory.INTERNAL_THREAD_POOL);
        System.out.println(sdFuture.get());
        MessagingFactory.INTERNAL_THREAD_POOL.shutdown();
    }

    private static String getTestString() throws MessagingEntityNotFoundException
    {
        throw new DateTimeParseException("test", "test", 1);
    }
```

Running this small example will also never compete, adjusting "getTestString()" to simply return something or throw the MessagingEntityNotFoundException finishes as expected. 
(Small bonus: Also this shows that currently MessagingFactory.INTERNAL_THREAD_POOL.shutdown() is never called e.g. ManagementClientAsync.close() also doesn't close it and this is consistently being reported as a thread leak on app shutdown. Let me know if you want a separate issue filed for this)

**Expected behavior**
Regardless of which exception is raised the returned CompletableFuture eventually completes and not hangs for days. This should be pretty easy to achieve by replacing "catch (MessagingEntityNotFoundException e)" with "catch (Exception e)" I believe.

**Screenshots**
<na>

**Setup (please complete the following information):**
 - Happened both on the RHEL7 Openshift Deployment as well as locally on Windows 10
- Eclipse
- 3.4.0 https://mvnrepository.com/artifact/com.microsoft.azure/azure-servicebus/3.4.0 and also 3.1.3

**Additional context**

**Information Checklist**
Kindly make sure that you have added all the following information above and checkoff the required fields otherwise we will treat the issuer as an incomplete report
- [x] Bug Description Added
- [x] Repro Steps Added
- [x] Setup information Added

**Comment by joshfree** [comment]
Thanks for filing this issue, @TheDevOps.  @yvgopal could you please assist in routing this azure-servicebus issue?

**Comment by yvgopal** [comment]
Resolved in version 3.6.4.
